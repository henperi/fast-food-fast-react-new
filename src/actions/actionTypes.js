export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SET_AUTH_USER = 'SET_AUTH_USER';
export const REMOVE_AUTH_USER = 'REMOVE_AUTH_USER';
export const SET_ERRORS = 'SET_ERRORS';
export const GET_MENU = 'GET_MENU';
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const UPDATE_CART_ITEM_QUANTITY = 'UPDATE_CART_ITEM_QUANTITY';
export const TOGGLE_CART_SLIDER = 'TOGGLE_CART_SLIDER';
export const FETCH_CART_ITEMS = 'FETCH_CART_ITEMS';
export const SET_VISIBLE_CART = 'SET_VISIBLE_CART';
