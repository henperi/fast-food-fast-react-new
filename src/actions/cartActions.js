import {
  ADD_TO_CART, REMOVE_FROM_CART, TOGGLE_CART_SLIDER, SET_CURRENT_CART,
} from './actionTypes';

export const setCurrentCart = payload => ({
  type: SET_CURRENT_CART,
  payload,
});

export const addToCart = (id, name, price) => ({
  type: ADD_TO_CART,
  id,
  name,
  price,
});

export const removeFromCart = id => ({ type: REMOVE_FROM_CART, id });

export const updateCartQty = (id, updates) => ({
  type: 'UPDATE_CART_ITEM_QUANTITY',
  updates,
  id,
});

export const emptyCart = () => {
  localStorage.removeItem('foodItems');
  return { type: 'EMPTY_CART' };
};

export const toggleCartSlider = () => ({ type: TOGGLE_CART_SLIDER });
