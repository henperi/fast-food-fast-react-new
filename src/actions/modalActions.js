// import { OPEN_MODAL } from './actionTypes';

// export const openModal = ({
//   id, cat, title, price, description,
// } = {}) => ({
//   type: OPEN_MODAL,
//   modal: {
//     isOpen: true,
//     id,
//     cat,
//     title,
//     price,
//     description,
//   },
// });

// export const emptyCart = () => ({ type: 'EMPTY_CART' });
