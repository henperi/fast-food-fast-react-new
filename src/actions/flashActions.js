import { SHOW_FLASH, REMOVE_FLASH } from './actionTypes';

export const showFlash = (data = {}) => {
  const { title = '', message = '', flashType = 'success' } = data;
  return {
    type: SHOW_FLASH,
    payload: {
      title,
      message,
      flashType,
    },
  };
};

export const removeFlash = () => ({
  type: REMOVE_FLASH,
});
