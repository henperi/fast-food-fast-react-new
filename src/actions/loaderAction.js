import { TOGGLE_LOADER } from './actionTypes';

const toggleLoader = () => ({
  type: TOGGLE_LOADER,
});

export default toggleLoader;
