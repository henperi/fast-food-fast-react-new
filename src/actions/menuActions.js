import { SET_MENU, SET_ERRORS, SET_FEATURED_MENU } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';

export const setMenu = (payload = {}) => ({
  type: SET_MENU,
  payload,
});

export const setFeaturedMenu = (payload = {}) => {
  const featuredMenu = payload.slice(0, 3);

  return {
    type: SET_FEATURED_MENU,
    payload: featuredMenu,
  };
};

export const getMenu = () => (dispatch) => {
  axiosInstance
    .get('/menu')
    .then((response) => {
      const { data } = response;
      dispatch({
        type: SET_MENU,
        payload: { menu: [...data.menu] },
      });
      dispatch(setFeaturedMenu(data.menu));
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const updateMenu = () => {};
