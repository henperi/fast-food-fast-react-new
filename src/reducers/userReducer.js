import { SIGNUP_USER, SET_AUTH_USER, REMOVE_AUTH_USER } from '../actions/actionTypes';

const userDefaultState = {
  isAuthenticated: false,
};

const userReducer = (state = userDefaultState, action) => {
  switch (action.type) {
    case SIGNUP_USER:
      return {
        ...action.payload,
      };
    case SET_AUTH_USER:
      return {
        ...state,
        isAuthenticated: !!action.payload,
        ...action.payload,
      };
    case REMOVE_AUTH_USER:
      return {
        isAuthenticated: false,
      };

    default:
      return state;
  }
};

export default userReducer;
