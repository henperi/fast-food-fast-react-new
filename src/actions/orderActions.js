import { SET_ERRORS } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';
import { emptyCart } from './cartActions';

const placeOrder = (cartItems = {}, handleModal) => (dispatch) => {
  console.log(cartItems);
  const orderData = {
    foodItems: cartItems,
  };
  axiosInstance
    .post('/orders', orderData)
    .then((response) => {
      const { data } = response;
      console.log(data);
      handleModal();
      dispatch(emptyCart());
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(errors);
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export default placeOrder;
