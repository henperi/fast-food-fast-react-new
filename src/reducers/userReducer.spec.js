/* eslint-disable no-undef */
import { SIGNUP_USER, SET_AUTH_USER, REMOVE_AUTH_USER } from '../actions/actionTypes';

// Reducer to be tested
import userReducer from './userReducer';

describe('userReducer', () => {
  const initialState = {
    isAuthenticated: false,
  };
  it('should return the default state if no action is specified', () => {
    const action = { type: '' };
    expect(userReducer(initialState, action)).toMatchSnapshot({});
    expect(userReducer(initialState, action)).toEqual(initialState);
  });

  // it('should update the state and set order history when type is SET_ORDER_HISTORY and action is specified', () => {
  //   const action = {
  //     type: SET_ORDER_HISTORY,
  //     payload: [{ id: 5, price: 500 }, { id: 3, price: 700 }],
  //   };
  //   expect(userReducer(initialState, action)).toMatchSnapshot({});
  //   expect(userReducer(initialState, action)).toEqual({
  //     orderHistory: [{ id: 5, price: 500 }, { id: 3, price: 700 }],
  //     orderedItems: [],
  //   });
  // });

  // it('should update the state and set order items when type is SET_ORDERED_ITEMS and action is specified', () => {
  //   const action = {
  //     type: SET_ORDERED_ITEMS,
  //     payload: [{ id: 5, price: 500 }, { id: 3, price: 700 }],
  //   };
  //   expect(userReducer(initialState, action)).toMatchSnapshot({});
  //   expect(userReducer(initialState, action)).toEqual({
  //     orderHistory: [],
  //     orderedItems: [{ id: 5, price: 500 }, { id: 3, price: 700 }],
  //   });
  // });

  // it('should update the state and remove an order history when type is REMOVE_ONE_ORDER_HISTORY and action is specified', () => {
  //   const action = {
  //     type: REMOVE_ONE_ORDER_HISTORY,
  //     id: 5,
  //   };
  //   expect(userReducer(initialState, action)).toMatchSnapshot({});
  //   expect(userReducer(initialState, action)).toEqual({ orderHistory: [], orderedItems: [] });
  // });
});
