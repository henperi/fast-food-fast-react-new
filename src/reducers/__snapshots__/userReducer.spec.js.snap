// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer should return the default state if no action is specified 1`] = `
Object {
  "isAuthenticated": false,
}
`;

exports[`userReducer should update the state and REMOVE_AUTH_USER when type is REMOVE_AUTH_USER and payload is specified 1`] = `
Object {
  "isAuthenticated": false,
  "user": Object {},
}
`;

exports[`userReducer should update the state and set SET_AUTH_USER when type is SET_AUTH_USER and action is specified 1`] = `
Object {
  "isAuthenticated": true,
  "user": Object {
    "name": 500,
    "user_id": 5,
  },
}
`;

exports[`userReducer should update the state and set SIGNUP_USER when type is SIGNUP_USER and action is specified 1`] = `
Object {
  "name": 500,
  "user_id": 5,
}
`;
